MobileDevProject/ViewController.swift

```diff
@@ -151,31 +158,131 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+//        DispatchQueue.global(qos: .background).async {
+            // If lineShapeLayer already exist, redraw the whole layer
+            if self.lineShapeLayer != nil {
+                self.lineShapeLayer.removeFromSuperlayer()
+            } else {
+                self.lineShapeLayer = CAShapeLayer()
+            }
+            let linePath = UIBezierPath()
+            
+            // if we want to draw multiple points just addLine to each new CGPoint
+            // we should want to but theres no easy way to work that out
+            linePath.move(to: start.coordinate)
+            linePath.addLine(to: end.coordinate)
+            self.lineShapeLayer.path = linePath.cgPath
+            
+            // line style
+            self.lineShapeLayer.strokeColor = UIColor.green.cgColor
+            self.lineShapeLayer.lineWidth = 1
+            // if we have multiple points to draw to in the future this sets the style of the corners
+            self.lineShapeLayer.lineJoin = kCALineJoinRound
+            //Add the line to the layer
+            self.mapView.layer.addSublayer(self.lineShapeLayer)
+            
+//            while true{
+                //make the function wait for half a second so there is enough time to draw
+//                usleep(1000000)
+ 
+                //remove last layer so you don't get circleShapeLayers on top of eachother
+        
+//            }
+//        }
+    }
+    
+    func updateCircle(fromPoint start: BeaconInfo, toPoint end: BeaconInfo) {
+        if self.circleShapeDrawn{
+            self.mapView.layer.sublayers?.remove(at: (self.mapView.layer.sublayers?.count)! - 2)
         }
-        let linePath = UIBezierPath()
         
-        // if we want to draw multiple points just addLine to each new CGPoint
-        // we should want to but theres no easy way to work that out
-        linePath.move(to: start)
-        linePath.addLine(to: end)
-        lineShapeLayer.path = linePath.cgPath
+        self.circleShapeLayer = CAShapeLayer();
+        
+        //Calculate where the circle needs to be drawn
+        let circleCordinates = self.calcXY(firstBeacon: start, secondBeacon: end)
+        let circlePath = UIBezierPath(arcCenter: circleCordinates, radius: CGFloat(7), startAngle: CGFloat(0), endAngle:CGFloat(Double.pi * 2), clockwise: true)
+        
+        self.circleShapeLayer.path = circlePath.cgPath
+        //change the fill color
+        self.circleShapeLayer.fillColor = UIColor.red.cgColor
+        //you can change the stroke color
+        self.circleShapeLayer.strokeColor = UIColor.red.cgColor
+        //you can change the line width
+        self.circleShapeLayer.lineWidth = 3.0
+        
+        //Add circle to the layer
+        self.mapView.layer.addSublayer(self.circleShapeLayer)
+        self.circleShapeDrawn = true
+    }
+    
+    func calcXY(firstBeacon: BeaconInfo, secondBeacon: BeaconInfo) -> CGPoint{
+
+        if beaconsArray.count <= 1 && cgPoint != nil{
```

Comment:
`beaconsArray.count <= 1 && cgPoint != nil` . What does this mean? Make a new variable `let condition = beaconsArray.count <= 1 && cgPoint != nil `. It will be easier to understand what the condition is.